[BOP-1341] Update authentication (#103)
Mirantis/mke-docs: content/docs/operations/authentication/_index.md

weight: 1
---

> MKE uses [Dex](https://dexidp.io/) for authentication. Dex behaves as a proxy
> between your MKE cluster and your authentication providers. Dex combines the
> configuration of multiple authentication providers into a single configuration
> and handles the complexity of the various protocols.

Reviewer: ```suggestion
MKE uses [Dex](https://dexidp.io/) for authentication. Dex serves as a proxy
between your MKE cluster and your authentication providers, combining the
configuration of multiple authentication providers into a single configuration
while also handling the complexity of the various protocols.
```

authentication. These dependencies differ, depending on which authentication
protocol you choose to deploy.

- **Identity Provider (IdP):** To use OIDC or SAML, you must configure an identity provider. For examples of how to use Okta as an authentication service provider for either of these protocols, refer to [OIDC](../../operations/authentication/oidc-providers/oidc) or [SAML](../../operations/authentication/saml-providers/saml).

- **LDAP Server:** To use LDAP, you must have an LDAP server configured. A setup example for an OpenLDAP server is available at [LDAP](../../tutorials/authentication-provider-setup/ldap).

## Configuration

You configure authentication for MKE through the `authentication` section
of the MKE configuration file.

> Authentication is is always enabled, however, the settings for each of the

Reviewer: ```suggestion
Authentication is always enabled, however, the settings for each of the
```

| `expiry`                                 | The section for the various expiry settings                                |
| `expiry.idTokens`                        | The lifetime of the ID tokens                                              |
| `expiry.authRequests`                    | The time frame that a code can be exchanged for a token                    |
| `expiry.deviceRequests`                  | The time frame in which users can authorize a device to receive a token    |
| `expiry.signingKeys`                     | The time period after which the signing keys are rotated                   |
| `expiry.refreshTokens`                   | The section for the various refresh token settings                         |
| `expiry.refreshTokens.validIfNotUsedFor` | Invalidate a refresh token if it's not used for the specified time         |
| `expiry.refreshTokens.absoluteLifetime`  | Absolute time frame of a refresh token                                     |
| `expiry.refreshTokens.disableRotation`   | Disable every-request rotation                                             |

> | `expiry.refreshTokens.reuseInterval`     | Interval to allow getting the same refresh token from the refresh endpoint |

Reviewer: So as not to screw up the table formatting, I am foregoing the Suggestion tool. I have a few asks, though:

1. Please put a period at the end of each description.
2. Please remove the 'The' from the first six descriptions and capitalize the word that follows.
3. Please change 'it's' on line 67 to 'it is'.
4. Should "every-request" be in code/command font?
5. Please change "Interval to allow getting" to "Interval for obtaining".

authentication. These dependencies differ, depending on which authentication
protocol you choose to deploy.

> - **Identity Provider (IdP):** To use OIDC or SAML, you must configure an
>   identity provider. Refer to [OIDC](../../operations/authentication/oidc-providers/oidc) or
>   [SAML](../../operations/authentication/saml-providers/saml) for examples
>   of using Okta as an authentication provider for either protocol.

Reviewer: ```suggestion
- **Identity Provider (IdP):** To use OIDC or SAML, you must configure an identity provider. For examples of how to use Okta as an authentication service provider for either of these protocols, refer to [OIDC](../../operations/authentication/oidc-providers/oidc) or [SAML](../../operations/authentication/saml-providers/saml).
```

> - **LDAP Server:** To use LDAP, you must have an LDAP server configured. An
>   example server on your local machine can be found at [LDAP](../../operations/authentication/ldap).

Reviewer: ```suggestion
- **LDAP Server:** To use LDAP, you must have an LDAP server configured. An
  example server on your local machine is available at [LDAP](../../operations/authentication/ldap).
```

Author: ```suggestion
- **LDAP Server:** To use LDAP, you must have an LDAP server configured. An example of setting up an OpenLDAP server can be found at [LDAP](../../operations/authentication/ldap).
```

Reviewer: Howzabout:

**LDAP Server:** To use LDAP, you must have an LDAP server configured. A setup example for an OpenLDAP server is available at [LDAP](../../operations/authentication/ldap).

## Configuration

You configure authentication for MKE through the `authentication` section
of the MKE configuration file.

Authentication is is always enabled, however, the settings for each of the
individual authentication protocols are disabled. To enable and install an
authentication protocol, set its `enabled` configuration option to `true`.

```yaml
authentication:
  enabled: true
  ldap:
    enabled: false
  oidc:
    enabled: false
  saml:
    enabled: false
```

### Expiry

> The `expiry` section of the configuration file allows you to set the expiration
> time for refresh and id tokens. These should be set in the format of number +
> time unit (e.g. `1h` for one hour).

Reviewer: ```suggestion
You can use the `expiry` section of the configuration file to set the expiration time for refresh and id tokens, in the format of number + time unit format. For example, `1h` to designate one hour.
```

```yaml
authentication:
  enabled: true
  expiry:
    refreshTokens: {}
```

> 

Reviewer: Need a short lead in sentence for the table. Something like, "The `expiry` configuration fields are defined in the following table:" Not sure of the correct word, but I am sure you are...maybe 'parameters', or 'options'?
